### PR TITLE
Adds pizza bomb

### PR DESCRIPTION
**Pull request by ghost**
The pizza bomb is a traitor item for four telecrystals. Activating it in your hand the first time will allow you to set the timer between 1 and 10 seconds. Activating it again will trigger the timer and blow up. You can attack the armed bomb with wirecutters to attempt to cut the correct wire and disarm the bomb. The bomb has the force of a minibomb.

Sprites for the pizza bomb are by me. This is mainly for fun.

deeda_bomb Pizza bomb sprite (there are sprites with wires cut)

### CODE REVIEW

**Comment by Iamgoofball** [comment]
all pizzas now immediately thrown down disposals

:+1:

**Comment by Cheridan** [comment]
What is a station-bounced radio

**Comment by Xhuis** [comment]
> These are mainly for fun

It's independent from tcomms, mainly intended for people looking to have private communications or just for fun

**Comment by Chocobro** [comment]
omg

**Comment by Cheridan** [comment]
Station bounced radios also work independently from tcomms

if you want private conversations just set it to a private channel

**Comment by Xhuis** [comment]
¯\_(ツ)_/¯

**Comment by Iamgoofball** [comment]
*¯\_(ツ)_/¯

also @Cheridan tcomms isn't private ever, players can just flick around the frequencies and hope they find the 2 traitors plotting to blow up the station

these walkie talkies are much more useful for private convos then the telecomms system

+ most tcomms scripts AIs load in have bullshit implemented so they can hear all frequencies at once

**Comment by Xhuis** [comment]
you can't hide it from me goofball
i know you tried to do the shrugging thingy too
but you failed
feel my pain~~~

**Comment by Nienhaus** [comment]
Pressing cancel on the walkie still sends a message, that has nothing in it.

**Comment by ACCount12** [comment]
WT are useless.

**Comment by ike709** [comment]
is it possible to also put pizza in the box to hide the bomb

**Comment by ike709** [comment]
also walkie talkies are needed for reliving childhood fun, but in spess

**Comment by Xhuis** [comment]
Hiding a pizza in there would be pointless since it would be a SUSPICIOUSLY BEEPING PIZZA
Will squash that @Nienhaus

**Comment by dannno** [comment]
the walkie talkies are entirely useless considering you can simply use a private channel on a station bounce radio (anyone who is cycling through those channels every round should be b& anyways)
the pizza seems... pointless? It just sounds like a minibomb that looks like a pizza. I don't get it.

**Comment by Iamgoofball** [comment]
why must there only be 1 option for communicating privately

why can't we have some variation in our spacemans

**Comment by Cheridan** [comment]
>durr there must be multiple items that do the exact same thing
>this is somehow variation even though they do the same thing

**Comment by Xhuis** [comment]
whoopsies didn't mean to leave that in there :^)

**Comment by dannno** [comment]
cheridan is right though...............................................

**Comment by Xhuis** [comment]
walkie-talkies are just for fun mainly
pizza bombs are gimmicky and for the fun of it
this entire pr is for the fun of it

**Comment by stapler2025** [comment]
heaven forbid we add fun things to the station guise

**Comment by dannno** [comment]
there's "for the fun of it"
and then there's stuff that's just extraneous and needless

**Comment by Xhuis** [comment]
like gavels?
:^)

**Comment by dannno** [comment]
you know you don't mean that

**Comment by Alek2ander** [comment]
Some comments on the walkie-talkie:

Communications should probably be limited to one z-level.
What do you think about distance-based message deterioration? (John Griff says, "H??p ?? ?og?e")
Not receiving messages while you have the talking-into-radio dialog box open (realism!)
How about making all other walkie-talkies emit a weaker Resonant Shriek as the emag action?

**Comment by Thunder12345** [comment]
If we're going realism, might as well go full realism. IIRC only one person can talk at a time on a set of walkie-talkies, so while someone has the dialog box open, nobody else can open it. ("Someone else is already transmitting") Maybe also a variety of frequencies available to work with, so if some dong holds the talk button all the time you can move to another frequency. Maybe use a screwdriver to jam the talk button down so you can dick over someone's conversation without having to keep the radio on you. (As soon as you take the radio out your hands it should stop trying to transmit and force you to reopen the box to speak)

**Comment by Alek2ander** [comment]
>IIRC only one person can talk at a time on a set of walkie-talkies

Not really. They'll just talk over each other and everyone else will have trouble discerning either's words (and the speakers won't hear each other).

But you can't talk and listen at the same time with classic walkie-talkies.

**Comment by Thunder12345** [comment]
From what I've seen it's one person talking at a time. Either way, it's supposed to be a really shitty way to talk privately, and being able to block it with one walkie-talkie and a screwdriver seems a better downside than just not being able to hear people talking at the same time.

**Comment by Jordie0608** [comment]
Changelog

**Comment by RemieRichards** [comment]
Just remove the walkie talkies ffs. they're pointless, as has been pointed out already.

File: code/game/objects/items/devices/walkie_talkie.dm [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+/obj/item/device/handheld_radio/proc/update_power()
+	off = !off
+	icon_state = "walkietalkie_[off]"
+	update_icon()
+
+/obj/item/device/handheld_radio/proc/broadcast_message(var/msg)
+	if(!msg)
+		return
+	for(var/obj/item/device/handheld_radio/R in world)
```
Comment (Miauw62) [inline code/game/objects/items/devices/walkie_talkie.dm]: eeewwwwww

File: code/game/objects/items/devices/walkie_talkie.dm [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	message = stripped_input(speaker, "", "Message to Send", "")
+	if(transmit_name)
+		if(!emagged)
+			message = "<b>([speaker.real_name])</b>: \"[message]\""
+		else
+			message = "<b>([emagged_name])</b>: \"[message]\""
+	else
+		message = "<b>(Radio ID #[id])</b>: \"[message]\""
+	speaker << "<span class='notice'>You broadcast: <b>\"[message]\"</b></span>"
+	broadcast_message(message)
```
Comment (Miauw62) [inline code/game/objects/items/devices/walkie_talkie.dm]: >not even a say(message)

**Comment by Miauw62** [comment]
awful :-1:

as others have said, walkie-talkies are completely obsolete and the code is pretty bad too.
the AI can't hear all frequencies, only frequencies for which busses/etc exist are picked up by the tcomms scripts afaik (@gun_hog confirmation?)

**Comment by dannno** [comment]
The thing is that you want to add a bunch of new code for an item, which functionally is already present in the form of station bounced radios.
The pizza bomb thing just seems unnecessary

**Comment by Dorsisdwarf** [comment]
Walky talkies provide secure comms. How is this so hard to get?

the MAGNI telecommscript (used by every powergaming AI and more) automatically detects and sends to the AI every possibly transmission you could send from a SBR.

**Comment by Miauw62** [comment]
/AS I HAVE JUST SAID, MAGNI TCOMMS DOES NOT CATCH FREQUENCIES WITHOUT A SERVER./ THAT IS WHY THIS IS USELESS.

**Comment by Miauw62** [comment]
If you look at this line in the manual:
https://github.com/GunHog/MAGNI-Tcomms/blob/master/Tcomms%20Manual.txt#L350
you will see that certain frequencies are only picked up if they are manually added to the telecomms machinery. e.g. normally any non-standard frequency is completely secure, even without having to resort to a snowflakey unstandardized device!

**Comment by RemieRichards** [comment]
@Dorsisdwarf >How is this so hard to get?

Stop being a patronising idiot, we get it, more so than you do apparently, because we know it's Pointless and have stated this repeatedly.

2 Maintainers and a Headcoder have weighed in that this is shit and unneeded.
Heck even Saycode-Man is unhappy with the say code attached to this PR.

There's also "X in World" crap which we don't do because its horrific.

tl;dr Remove Walkie Talkies if you want any of us to take this PR seriously.

**Comment by Xhuis** [comment]
k

**Comment by Xhuis** [comment]
I left the code commented out rather than deleting so if I ever decide to come back to it (or someone else) I have something to work from

**Comment by RemieRichards** [comment]
We don't keep commented code, Simply keep the branch, git will handle the history of it.

**Comment by Miauw62** [comment]
Yeah, just make a commit to remove the code and keep a link to it
somewhere. Git is basically made for that stuff.
Op 28-mrt.-2015 16:45 schreef "Remie Richards" <notifications@github.com>:

**Comment by Xhuis** [comment]
mmkay

**Comment by Xhuis** [comment]
https://www.youtube.com/watch?v=KolfEhV-KiA

**Comment by Xhuis** [comment]
Squashed commits

**Comment by Xhuis** [comment]
That is Travis' fault, not mine

**Comment by Xhuis** [comment]
I have literally no idea why there are two commits named 'a'. There's only one that's supposed to be named 'a'. Whatever.

Okay well the merge conflicts are fixed. Would appreciate a merge when it has 24 hours so there's no more issues with it unless someone has a problem with it.

**Comment by Supermichael777** [comment]
need to be able to wire mousetrap to the pizza box so it goes boom wen the pizza is removed.

**Comment by Xhuis** [comment]
There isn't a pizza in there to begin with. The way it works is when you open the box it triggers the timer.

**Comment by Chocobro** [comment]
but surely the ability should be there
as like a prize "hooray you disabled the bomb have a pizza" or "the bomb blew up you lost the pizza 0/10" or something
there is a chance that i am entirely drunk right now

**Comment by Xhuis** [comment]
A pizza isn't in the box to begin with though. Pizzas can't fabricate themselves from nothingness.

**Comment by Jordie0608** [comment]
When disarming the bomb the cables, bombcore and pizza box appear beneath the mob not the tile that the pizza bomb was on.

Also changelog.

File: code/game/machinery/syndicatebomb.dm [hunk 3]
```diff
@@ -274,6 +274,17 @@
...
 
+/obj/item/weapon/bombcore/miniature
+	name = "small bomb core"
+	w_class = 2
+
+/obj/item/weapon/bombcore/miniature/detonate()
+	if(adminlog)
+		message_admins(adminlog)
+		log_game(adminlog)
+	explosion(src.loc,1,2,4,flame_range = 2) //Identical to a minibomb
```
Comment (Jordie0608) [inline code/game/machinery/syndicatebomb.dm]: The explosion proc already logs to admins: https://github.com/tgstation/-tg-station/blob/master/code/game/objects/explosion.dm#L45

Aside from that; there's no actual message here, all these two procs would do it message and log to the admins "adminlog".

Comment (Alek2ander) [inline code/game/machinery/syndicatebomb.dm]: The explosion proc only logs that an explosion occured. It's usually useful to know who and/or what caused it.

Comment (Jordie0608) [inline code/game/machinery/syndicatebomb.dm]: The code here doesn't do that either; it would need to log user upon priming, activation and non-timed detonation of the bomb for that.

**Comment by Xhuis** [comment]
@Jordie0608 I made the player unequip the pizza bomb to prevent issues like having it in their backpack and disarming it, which would result in all the items disappearing. I felt it's a minor thing, if you really dislike it, though, I can find a workaround.

Will do changelog in a bit.

**Comment by Xhuis** [comment]
This is basically complete, if there's a problem with merging let me know and I'll deal with it.

**Comment by Jordie0608** [comment]
Fair enough.

**Comment by Xhuis** [comment]
Merge-ready

**Comment by Jordie0608** [comment]
Could you add logging as requested above?

**Comment by Xhuis** [comment]
Currently on hiatus until Monday, but I will fix the issues on all my PRs that need fixing once I come back.

**Comment by Xhuis** [comment]
Logging and messaging have been added

File: code/game/objects/items/devices/pizza_bomb.dm [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+/obj/item/device/pizza_bomb/attack_self(mob/user)
+	if(disarmed)
+		user << "<span class='notice'>\The [src] is disarmed.</span>"
+		return
+	if(!timer_set)
+		name = "pizza bomb"
+		desc = "It seems inactive."
+		icon_state = "pizzabox_bomb"
+		timer_set = 1
+		timer = (input(user, "Set a timer, from one second to ten seconds.", "Timer", "[timer]") as num) * 10
```
Comment (phil235) [inline code/game/objects/items/devices/pizza_bomb.dm]: There needs to be an adjacency and stat check after this

Comment (Miauw62) [inline code/game/objects/items/devices/pizza_bomb.dm]: Use CanUseTopic
Op 7-apr.-2015 19:20 schreef "phil235" <notifications@github.com>:

**Comment by Xhuis** [comment]
Anything wrong with merging this?

File: code/game/machinery/syndicatebomb.dm [hunk 5]
```diff
@@ -274,6 +274,14 @@
 /obj/item/weapon/bombcore/badmin/explosion/detonate()
 	explosion(get_turf(src),HeavyExplosion,MediumExplosion,LightExplosion, flame_range = Flames)
 
+/obj/item/weapon/bombcore/miniature
+	name = "small bomb core"
+	w_class = 2
+
+/obj/item/weapon/bombcore/miniature/detonate()
+	explosion(src.loc,1,2,4,flame_range = 2) //Identical to a minibomb
+	qdel(src)
```
Comment (phil235) [inline code/game/machinery/syndicatebomb.dm]: why did you remove the if(adminlog) stuff?

Comment (Jordie0608) [inline code/game/machinery/syndicatebomb.dm]: @phil235 https://github.com/tgstation/-tg-station/pull/8582#discussion_r27460171
